[COP] T1 Sea NM System Mechanics

The-Aerec opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Adds retail like mechanics to the following mobs and their minions:

Ix'DRK
Ix'DRG
Ix'MNK
Jailer of Fortitude
Jailer of Faith
Jailer of Temperance

The mechanics / immunities were taken from the following sources:

Ix'DRK:
https://youtu.be/j7k9HsChq4c

Ix'DRG:
https://www.youtube.com/watch?v=1h4BDu50zho
https://youtu.be/Ug2Y95qvG8Q

Ix'MNK:
https://youtu.be/k413vn8SbrU
https://youtu.be/7YtDok1lhvo

Jailer of Fortitude:
https://youtu.be/V7jd3b1kzkg
https://youtu.be/-hfYXA4wAj4

Jailer of Faith:
https://youtu.be/eIKrocH0WyM

Jailer of Temperance:
https://youtu.be/T2OPAQOv9gg
https://www.youtube.com/watch?v=wntIOcnQoAk

All the above videos were taken from the FFXI Captures discord and have associated Packet logs as well. (Which I havn't posted to keep this as clean as possible)

If information was missing it was filled in via:
https://www.bg-wiki.com/ffxi/Category:Sea_NM_System

## Steps to test these changes

Spawn any of these mobs and fight to observe mechanics.

The-Aerec committed:
Ix'DRG Mechanics

Co-Authored-by: Kravy <anthony925@comcast.net>

The-Aerec committed:
Ix'DRG Mechanics

Co-Authored-by: Kravy <anthony925@comcast.net>

TeoTwawki commented:
Can the Aern crash fix be split out to its own pr? There's likely to be a lot to review here when you are finished with your draft and I'd like to get the crash fix in sooner than that.

The-Aerec commented:
> 

Can do

The-Aerec committed:
Sea NM Systems T1

Co-Authored-By: Kravy <anthony925@comcast.net>

The-Aerec committed:
Sea NM Systems T1

Co-Authored-By: Kravy <anthony925@comcast.net>

zach2good commented:
Commits look good, I don't know anything about the game, so someone else will have to review the actual content 👁️ 👁️

TeoTwawki commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Ixaern_MNK.lua:
```diff
@@ -8,53 +8,94 @@ require("scripts/globals/status")
+        end
+    end
+    -- slightly delay adding local var to avoid adding bracers to Ix'Mnk while Hundred Fists is still active.
+    mob:timer(3000, function(mobArg)
```
I would rather we set vars each for both 2hrs it track them each than use timer, as onMobFight will continue to call this helper before the variable is set here and it is _possible_ to get edge cases seemingly at random where mob Ai queues up the moves repeatedly. onMobFight will exec every 400ms, slightly below half a second. It may work as you have it during your testing could be safer. I would still keep the bracer variable, but use the 2hr vars when deciding to set it if that makes sense.

The-Aerec replied:
I did this in the original implementation. (Wings side) And it always just still two hour'd, wings just didn't update fast enough to catch the local var in time.

TeoTwawki replied:
likely in wings it was altered to get around several similar timing bugs. Old dsp used to update mobs on a much slower tick rate. I would wager in wings that change got rolled back (it was the reason colibri were derping out when I tried to code them)

TeoTwawki replied:
I am apparently illterate today, read BracerMode and enableBracers as same variable.

TeoTwawki commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Ixaern_MNK.lua:
```diff
@@ -8,53 +8,94 @@ require("scripts/globals/status")
-        SetDropRate(4398, 1851, 0)
-        SetDropRate(4398, 1901, chance * 10) -- Vice of Antipathy
+        SetDropRate(2845, 1851, 0)
+        SetDropRate(2845, 1901, chance * 10) -- Vice of Antipathy
```
@zach2good is this still the pattern we want for drop lists that change mid fight? I know we have several ways that work but I no longer remember what the gold standard was.

claywar replied:
Would like to see xi.items used here as well if this is the route we're going

The-Aerec replied:
Just changing it so it works, let me know whatever style you want. The droplist it was referencing was incorrect.

TeoTwawki commented on scripts/zones/The_Garden_of_RuHmet/mobs/Jailer_of_Faith.lua:
```diff
@@ -7,27 +7,114 @@ mixins = {require("scripts/mixins/job_special")}
+    end)
+end
+
+entity.onMagicCastingCheck = function(mob, target, spell)
```
did you mean `onMobMagicPrepare()` ?

The-Aerec replied:
When converting to LSB base I actually didn't find any use of that specific function. So I looked around and found that Promathia's original LSB implementation used: onMagicCastingCheck so I went with that.

The-Aerec replied:
See from Promathia_2:

```lua

entity.onMagicCastingCheck = function(mob, target, spell)
    if math.random() > 0.75 then
        return 219
    else
        return 218
    end
end

```

TeoTwawki replied:
quote possibly a mistake there, or severely undocumented functionality. 

onMobMagicPrepare tho, seems to be used precisely the way you want in many existing places:
https://github.com/LandSandBoat/server/search?p=2&q=onMobMagicPrepare

The-Aerec replied:
I'll adjust both then.

The-Aerec committed:
IxMNK Drop Adjustments

Adds IxMNK dropchanges to xi.items

The-Aerec committed:
IxMNK Pet Adjustments and SpellCall fixes

The-Aerec committed:
IxMNK Pet Adjustments and SpellCall fixes

The-Aerec committed:
Ix'DRG Mechanics

Co-Authored-by: Kravy <anthony925@comcast.net>
Sea NM Systems T1

Co-Authored-By: Kravy <anthony925@comcast.net>
IxMNK Drop Adjustments

Adds IxMNK dropchanges to xi.items
IxMNK Pet Adjustments and SpellCall fixes

The-Aerec committed:
Ix'DRG Mechanics

Co-Authored-by: Kravy <anthony925@comcast.net>
Sea NM Systems T1

Co-Authored-By: Kravy <anthony925@comcast.net>
IxMNK Drop Adjustments

Adds IxMNK dropchanges to xi.items
IxMNK Pet Adjustments and SpellCall fixes

The-Aerec committed:
Merge remote-tracking branch 'origin/Sea_NM_System_pt1' into Sea_NM_System_pt1

The-Aerec committed:
Commit Squash

Co-Authored-By: Kravy <anthony925@comcast.net>

The-Aerec committed:
Merge remote-tracking branch 'origin/Sea_NM_System_pt1' into Sea_NM_System_pt1

The-Aerec committed:
Merge remote-tracking branch 'origin/Sea_NM_System_pt1' into Sea_NM_System_pt1

The-Aerec committed:
Merge remote-tracking branch 'origin/Sea_NM_System_pt1' into Sea_NM_System_pt1

The-Aerec committed:
Merge remote-tracking branch 'LSB_Main/base' into Sea_NM_System_pt1

The-Aerec committed:
Conflict Resolve

TeoTwawki commented on scripts/zones/Empyreal_Paradox/mobs/Promathia_2.lua:
```diff
@@ -67,7 +67,7 @@ entity.onSpellPrecast = function(mob, spell)
-entity.onMagicCastingCheck = function(mob, target, spell)
+entity.onMobMagicPrepare = function(mob, target, spell)
     if math.random() > 0.75 then
         return 219
```
I know its comment and meteor from memory, but lets comment them for ppl who won't know what the numbers are

The-Aerec replied:
Sure this was just an un-related fix that you and I had talked about but i'll add the 'what this is' to the comments

TeoTwawki replied:
ty

TeoTwawki commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Ixaern_MNK.lua:
```diff
@@ -5,56 +5,98 @@
+        if pet:isSpawned() then
+            pet:AnimationSub(2)
+            pet:addMod(xi.mod.ATT, 200)
+            pet:setMod(xi.mod.DELAY, 2100)
```
that modifier does not work how you think it does. please stop using it. We need to do some reworking in regard to delay setting because the various ways to set/alter delays are inconsistent and causing confusion. `xi.mod.DELAY` is used for % reductions (see blitzers roll). You want the function `mob:setDelay(2100)`  instead (I think) for the input you are trying to give this.

A similar mixup happened on ladybugs and will need fixed.

TeoTwawki commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Jailer_of_Temperance.lua:
```diff
@@ -21,26 +47,37 @@ entity.onMobSpawn = function(mob)
-    for n =1, #xi.magic.specificDmgTakenMod, 1 do
-        mob:setMod(xi.magic.specificDmgTakenMod[n], 10000)
-    end
+    mob:setMod(xi.mod.DMGMAGIC, -99)
```
99% ?

I'm tired so maybe I am missing something but I thought you wanted 100% for no damage per the comment. I vaguely recall a reason in the past for someone using the element mods instead of straight magic damage reduction but I'd need to test a non elemental spell on retail to know for sure, and its their fault for not explaining if they really needed it that way - so it can be like you got it just make sure the value is what you needed.

The-Aerec committed:
Comment Addressing

zach2good commented on scripts/zones/Empyreal_Paradox/mobs/Promathia_2.lua:
```diff
@@ -68,11 +68,11 @@ entity.onSpellPrecast = function(mob, spell)
+entity.onMobMagicPrepare = function(mob, target, spell)
     if math.random() > 0.75 then
-        return 219
+        return 219 -- comet
```
`xi.magic.spell.COMET` etc.

zach2good commented on scripts/zones/Empyreal_Paradox/mobs/Promathia_2.lua:
```diff
@@ -68,11 +68,11 @@ entity.onSpellPrecast = function(mob, spell)
+        return 219 -- comet
     else
-        return 218
+        return 218 -- meteor
```
xi.magic.spell.METEOR

zach2good commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Jailer_of_Temperance.lua:
```diff
@@ -93,6 +141,27 @@ entity.onMobFight = function(mob)
+
+entity.onMobWeaponSkill = function(target, mob, skill)
+    local skillID = skill:getID()
+    if skillID == 1465 then
```
Comment what this is, even if its somewhat clear in the code below

zach2good commented on scripts/zones/The_Garden_of_RuHmet/mobs/Ixaern_DRGs_Wynav.lua:
```diff
@@ -38,6 +39,20 @@ entity.onMobMagicPrepare = function(mob, target, spellId)
+    if ixdrg:isSpawned() then
+        local mobId = mob:getID()
+        local pet = GetMobByID(mobId - 1)
+        switch (mobId): caseof
```
👍

zach2good commented on scripts/zones/The_Garden_of_RuHmet/mobs/Jailer_of_Faith.lua:
```diff
@@ -7,27 +7,114 @@ mixins = {require("scripts/mixins/job_special")}
+end
+
+entity.onMobMagicPrepare = function(mob, target, spell)
+    local rnd = math.random()
+    if rnd < 0.5 and mob:hasStatusEffect(xi.effect.MANAFONT) == true then -- quake II replaces existing earth damage spells during manafont
+        return 211 -- quake II casted exclusively while Manafont is active.
+    elseif rnd < 0.25 then
+        return 162 -- stone iv
+    elseif rnd < 0.5 then
+        return 191 -- stonega iii
+    elseif rnd < 0.75 then
+        return 365 -- breakga'
+    else
+        return 357 -- slowga
```
These can all use the spell enum names

zach2good commented on scripts/zones/The_Garden_of_RuHmet/mobs/Kfghrah_BLM.lua:
```diff
@@ -45,6 +45,17 @@ entity.onMobFight = function(mob, target)
 end
 
+entity.onMagicCastingCheck = function(mob, target, spell)
+    local rnd = math.random()
+    if rnd < 0.2 then
+        return 274 -- sleepga II
+    elseif rnd < 0.6 then
+        return 254 -- blind
+    else
+        return 232 -- bio III
+    end
```
Here too

zach2good commented on scripts/zones/The_Garden_of_RuHmet/mobs/Kfghrah_WHM.lua:
```diff
@@ -42,6 +42,22 @@ entity.onMobFight = function(mob, target)
+end
+
+entity.onMagicCastingCheck = function(mob, target, spell)
+    local rnd = math.random()
```
and again

The-Aerec committed:
Merge branch 'base' into Sea_NM_System_pt1

The-Aerec committed:
Merge remote-tracking branch 'LSB/base' into Sea_NM_System_pt1

The-Aerec commented:
Updated to fix conflicts

WinterSolstice8 commented on scripts/zones/The_Garden_of_RuHmet/mobs/Jailer_of_Faith.lua:
```diff
@@ -7,27 +7,114 @@ mixins = {require("scripts/mixins/job_special")}
+    })
     -- Change animation to open
-    mob:setAnimationSub(2)
+    mob:AnimationSub(2)
```
should be `mob:setAnimationSub(2)`

WinterSolstice8 commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Ixaern_MNK.lua:
```diff
@@ -5,56 +5,98 @@
     end
+    -- Ix'Mnk will not visually add Bracers while under the effect of Hundred Fists
+    if not mob:hasStatusEffect(xi.effect.HUNDRED_FISTS) and mob:getLocalVar("enableBracers") == 1 then
+        mob:AnimationSub(2) -- Bracers
```
should be `mob:setAnimationSub(2)`

WinterSolstice8 commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Ixaern_MNK.lua:
```diff
@@ -5,56 +5,98 @@
+        mob:AnimationSub(2) -- Bracers
+        mob:setDelay(1700)
+    else
+        mob:setDelay(0)
```
This is setting delay to 0 without bracers and no HF, I don't think this is right. I fought it on retail and it seemed to have the normal 480 delay without bracers. Does HF alone not correct the delay?

WinterSolstice8 reviewed (changes requested):
I'm in the middle of verifying this on retail (as last I did this was ages ago) + reviewing what's on LSB, I hope to get the review done by the weekend

WinterSolstice8 commented:
p.s. at some point this snuck in and I can't comment on the file
![image](https://user-images.githubusercontent.com/60417494/192075233-63ebfb65-3d14-439c-931a-fe9e0deff8fa.png)

WinterSolstice8 commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Jailer_of_Temperance.lua:
```diff
@@ -9,9 +9,35 @@ require("scripts/globals/magic")
 -----------------------------------
 local entity = {}
 
+local chargeOptic = function(mob)
```
Looks like this function's logic makes Temperance spam optic induration infinitely after a single use

I tested this locally:
```
local chargeOptic = function(mob)
    mob:SetAutoAttackEnabled(false)
    mob:SetMobAbilityEnabled(false)

    mob:timer(3000, function(mobArg)
        mobArg:useMobAbility(1465) -- JoT will use another Optic Induration shortly after using the first one.
    end)
end
```

This paired with an update to `entity.onMobWeaponSkill`, 

```
entity.onMobWeaponSkill = function(target, mob, skill)
    local skillID = skill:getID()
    if skillID == 1465 then -- On Optic Induration Do this
        local opticCounter = mob:getLocalVar("opticCounter")

        if opticCounter == 0 then
            mob:setLocalVar("opticCounter", 1)
            chargeOptic(mob)
        else
            mob:setLocalVar("opticCounter", 0)
            mob:SetAutoAttackEnabled(true)
            mob:SetMobAbilityEnabled(true)
        end
    end
end
```
fixes the problem

WinterSolstice8 commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Jailer_of_Temperance.lua:
```diff
@@ -21,26 +47,37 @@ entity.onMobSpawn = function(mob)
-    for n =1, #xi.magic.specificDmgTakenMod, 1 do
-        mob:setMod(xi.magic.specificDmgTakenMod[n], 10000)
-    end
+    mob:setMod(xi.mod.DMGMAGIC, -100)
```
should be `-1000`

WinterSolstice8 commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Jailer_of_Temperance.lua:
```diff
@@ -21,26 +47,37 @@ entity.onMobSpawn = function(mob)
+        if
+            act == xi.act.MOBABILITY_START or
+            act == xi.act.MOBABILITY_USING or
+            act == xi.act.MOBABILITY_FINISH
```
should add a check for `mob:getLocalVar("opticCounter") == 1` here too so it doesnt change forms between the magic double induration as well

WinterSolstice8 committed:
[lua] Adjust JoT optic induration logic, magic damage immunity, form change "busy" logic

* Change non-existant AnimationSub calls to setAnimationSub

WinterSolstice8 committed:
[lua] Add H.Q. Aern Organ definition and use in ixaern mnk QM

WinterSolstice8 committed:
[lua] Remove Ix'Aern MNK setdelay calls as the HF status effect handles that

TeoTwawki commented on scripts/zones/Grand_Palace_of_HuXzoi/mobs/Ixaern_MNK.lua:
```diff
@@ -5,56 +5,102 @@
     qm:setLocalVar("[SEA]IxAern_DropRate", 0)
 
     mob:setAnimationSub(1) -- Reset the subanim - otherwise it will respawn with bracers on. Note that Aerns are never actually supposed to be in subanim 0.
+    mob:setLocalVar("enableBracers", 0)
```
localvars are already wiped on spawn automatically, so this isn't needed

TeoTwawki commented on scripts/zones/The_Garden_of_RuHmet/mobs/Ixaern_DRG.lua:
```diff
@@ -6,25 +6,69 @@ local ID = require("scripts/zones/The_Garden_of_RuHmet/IDs")
+        battleTime - changeTime > 60
+    then -- goes into bracer mode for 30 seconds, every 60 seconds.
+        mob:setAnimationSub(2) -- bracer mode
+        mob:setMod(xi.mod.DELAY, 1000) -- increase speed in bracer mode
```
this is a bad way to mod mobs attack speed

TeoTwawki commented on scripts/zones/The_Garden_of_RuHmet/mobs/Ixaern_DRG.lua:
```diff
@@ -6,25 +6,69 @@ local ID = require("scripts/zones/The_Garden_of_RuHmet/IDs")
+        battleTime - changeTime > 30
+    then
+        mob:setAnimationSub(1)
+        mob:setMod(xi.mod.DELAY, 0) -- back to normal attack speed
```
this too

if its speed is changing, its likely some type of haste - probably ability haste

TeoTwawki commented on scripts/zones/The_Garden_of_RuHmet/mobs/Jailer_of_Faith.lua:
```diff
@@ -7,27 +7,114 @@ mixins = { require("scripts/mixins/job_special") }
+    mob:setLocalVar("RangedDamage", 0)
+    mob:setLocalVar("BreathDamage", 0)
+    mob:delMod(xi.mod.ATTP, 10)
+    mob:setMod(xi.mod.HTH_SDT, 1250)
+    mob:setMod(xi.mod.SLASH_SDT, 1250)
+    mob:setMod(xi.mod.PIERCE_SDT, 1250)
+    mob:setMod(xi.mod.IMPACT_SDT, 1250)
+    mob:setMod(xi.mod.DMGPHYS, 112)
```
we've got every type of physical damage -and-  the all phys dmg mod? I'm confused on the intent.

WinterSolstice8 replied:
My guess? copy paste from JoT. Probably only needs DMGPHYS

TeoTwawki commented on scripts/zones/The_Garden_of_RuHmet/mobs/Jailer_of_Faith.lua:
```diff
@@ -7,27 +7,114 @@ mixins = { require("scripts/mixins/job_special") }
+
+entity.onMobMagicPrepare = function(mob, target, spell)
+    local rnd = math.random()
+    if rnd < 0.5 and mob:hasStatusEffect(xi.effect.MANAFONT) == true then -- quake II replaces existing earth damage spells during manafont
```
this code doesn't accomplish what the comment says, only has a 50% ish chance of selecting quake, and then will still move on to those other earth spells. Either the comment ain't right or the logic ain't right.

WinterSolstice8 replied:
```
[21:23:53] [Jailer・of・Faith] Manafont → Jailer・of・Faith
[21:23:57] [Jailer・of・Faith] Quake II → [Redacted]
[21:23:57] ([Redacted]) open, was closed
[21:24:04] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:24:08] [Jailer・of・Faith] Slowga → [Redacted]
[21:24:11] Jailer・of・Faith Slowga → [Redacted] (Resists)
[21:24:12] [[Redacted]] Flash → Jailer・of・Faith
[21:24:13] [Redacted] Flash → Jailer・of・Faith (Flash)
[21:24:14] [Jailer・of・Faith] missed → [Redacted]
[21:24:14] [Jailer・of・Faith] Quake II → [Redacted]
[21:24:19] The Jailer of Faith is no longer afflicted with Flash.
[21:24:22] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:24:26] [Jailer・of・Faith] Breakga → [Redacted]
[21:24:31] Jailer・of・Faith Breakga → [Redacted] (Resists)
[21:24:34] [Jailer・of・Faith] Stonega III → [Redacted]
[21:24:39] [Jailer・of・Faith] 39 Stonega III → [Redacted]
[21:24:41] [Jailer・of・Faith] missed → [Redacted]
[21:24:42] [Jailer・of・Faith] Quake II → [Redacted]
[21:24:50] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:24:54] [Jailer・of・Faith] Stone IV → [Redacted]
[21:24:59] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:26:23] [Jailer・of・Faith] Manafont → Jailer・of・Faith
[21:26:23] [Jailer・of・Faith] Viscid Nectar → [Redacted]
[21:26:24] [Jailer・of・Faith] Viscid Nectar (Miss) → [Redacted]
[21:26:31] [Jailer・of・Faith] Slowga → [Redacted]
[21:26:33] Jailer・of・Faith Slowga → [Redacted] (Resists)
[21:26:37] [Jailer・of・Faith] Quake II → [Redacted]
[21:26:44] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:26:47] [2] [Jailer・of・Faith] missed → [Redacted]
[21:26:48] [Jailer・of・Faith] Stone IV → [Redacted]
[21:26:53] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:26:53] {2}: Cornelia and Sakura Afterglow effect wears off.
[21:26:57] [Jailer・of・Faith] Stonega III → [Redacted]
[21:27:02] [Jailer・of・Faith] 162 Stonega III → [Redacted]
[21:27:05] [Jailer・of・Faith] Stone IV → [Redacted]
[21:27:10] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:27:13] [Jailer・of・Faith] missed → [Redacted]
[21:27:14] [Jailer・of・Faith] Quake II → [Redacted]
[21:27:22] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:27:24] [Jailer・of・Faith] Manafont → Jailer・of・Faith
[21:27:25] [Jailer・of・Faith] Quake II → [Redacted]
[21:27:33] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:27:36] [Jailer・of・Faith] Stonega III → [Redacted]
[21:27:41] [Jailer・of・Faith] 39 Stonega III → [Redacted]
[21:27:44] [Jailer・of・Faith] missed → [Redacted]
[21:27:45] [Jailer・of・Faith] Quake II → [Redacted]
[21:27:52] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:27:56] [Jailer・of・Faith] Stone IV → [Redacted]
[21:28:01] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:28:05] [Jailer・of・Faith] Breakga → [Redacted]
[21:28:10] Jailer・of・Faith Breakga → [Redacted] (Resists!)
[21:28:12] [Jailer・of・Faith] missed → [Redacted]
[21:28:13] [Jailer・of・Faith] Stone IV → [Redacted]
[21:28:19] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:28:23] [Jailer・of・Faith] Stonega III → [Redacted]
```

Not a whole lot of data, but with 3 manafonts it seems that it doesn't always choose Quake 2, but it might choose Quake 2 immediately after manafont.

TeoTwawki replied:
> doesn't always choose Quake 2

So comment not _exactly_ right then

probably higher priority of multiple of its "better" damage spells but not 100% on any

WinterSolstice8 committed:
[lua] Add H.Q. Euvhi organ to items.lua, update JoF behavior to better match retail

TeoTwawki commented on scripts/zones/The_Garden_of_RuHmet/mobs/Jailer_of_Faith.lua:
```diff
@@ -97,17 +110,8 @@ entity.onMobFight = function(mob)
-    else
-        return xi.magic.spell.SLOWGA
+    if mob:hasStatusEffect(xi.effect.MANAFONT) and math.random() < 0.5 then
+        return xi.magic.spell.QUAKE_II
```
might still need to `return spell:getID()` when quake doesn't get hit, not sure.

WinterSolstice8 replied:
I checked, it falls back to the regular spell list which was identical. Works perfect in testing

TeoTwawki replied:
awesome, thanks Winter

WinterSolstice8 committed:
[lua] Adjust JoF extra damage on open/close to match data from retail

The-Aerec committed:
Merge remote-tracking branch 'origin/Sea_NM_System_pt1' into Sea_NM_System_pt1
